### PR TITLE
De-Scattering of superblock and vote inclusion in blocks.

### PR DESCRIPTION
**Pull request by tomasbrod**
* Removed a lot of `extern` declarations.
* Added declaration to headers, so code in main can be used in miner
* moved superblock creation to miner `AddNeuralContractOrVote`
* removed unrelated code from `SerializeBoincBlock`
* moved serializing of boincblock to `SignStakeBlock`, where the signing is less hacky
* superblock or neural vote is created in `AddNeuralContractOrVote`
* `qtGetNeuralHash` is no longer used, the vote hash is now calculated in c++
* removed now unused function `GetNeuralNetworkSuperBlock` +1 from main.cpp

### CODE REVIEW

File: src/miner.cpp [hunk 1]
```diff
@@ -647,7 +648,60 @@ bool SignStakeBlock(CBlock &block, CKey &key, vector<const CWalletTx*> &StakeInp
...
+int AddNeuralContractOrVote(const CBlock &blocknew, MiningCPID &bb)
+{
+    const std::string sb_contract;
+
+    if(OutOfSyncByAge())
+        return printf("AddNeuralContractOrVote: Out Of Sync\n");
+
+    /* Retrive the neural Contract */
+    #if defined(WIN32) && defined(QT_GUI)
+        const std::string sb_contract = qtGetNeuralContract("");
```
Comment (tomasbrod) [inline src/miner.cpp]: redefinition here

**Comment by denravonska** [comment]
I can't find anything inherently wrong with this but it's hard to follow when the code is moved (just the nature of the diff tool). I'll merge this so we can test it out in testnet during the pre-release test phase.
